src/main/scala/scala/tools/refactoring/util/SourceHelpers.scala

```diff
@@ -0,0 +1,44 @@
+package scala.tools.refactoring.util
+
+import scala.annotation.tailrec
+import scala.math.min
+
+object SourceHelpers {
+  def isRangeWithin(text: String, selection: SourceWithSelection): Boolean = {
+    if (selection.length > text.length || selection.source.length < text.length) {
+      false
+    } else {
```

Comment:
Because this method works the other way round (I've added some documentation in order to clarify); it might still be possible to implement it more concisely though, especially the inner loop. I decided to do this in a rather low level manner for performance reasons, but I guess it wont matter anyway.